File: ios/ExpoAiContextShieldModule.swift
import ExpoModulesCore

public class ExpoAiContextShieldModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAiContextShield")

    OnStartObserving {
      NotificationCenter.default.addObserver(
        forName: UIApplication.userDidTakeScreenshotNotification,
        object: nil,
        queue: .main
      ) { _ in
        self.sendEvent("onScreenshotDetected")
      }
    }

    View(ExpoAiContextShieldView.self) {
      Prop("isSensitive") { (view: ExpoAiContextShieldView, isSensitive: Bool) in
        view.setSensitive(isSensitive)
      }
    }
  }
}

File: android/src/main/java/expo/modules/aishield/ExpoAiContextShieldModule.kt
package expo.modules.aishield

import android.view.WindowManager
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoAiContextShieldModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoAiContextShield")

    Events("onScreenshotDetected")

    Function("setVisualSecurity") { secure: Boolean ->
      ExpoAiContextShieldView.updateWindowSecurity(appContext, if (secure) 1 else -1)
    }

    View(ExpoAiContextShieldView::class) {
      Prop("isSensitive") { view: ExpoAiContextShieldView, isSensitive: Boolean ->
        view.setSensitive(isSensitive)
      }
    }
  }
}